runtime/src/snapshot_utils.rs

```diff
@@ -422,6 +422,21 @@ pub enum VerifySlotDeltasError {
     BadSlotHistory,
 }
 
+/// Creates directories if they do not exist, and canonicalizes the paths.
+pub fn create_and_canonicalize_directories(directories: &[PathBuf]) -> Result<Vec<PathBuf>> {
```

Comment:
wasn't super happy about this returning a `SnapshotError`, since it's only somewhat related to snapshots. Somewhat of a hack because I want to keep info about which path caused the error which afaik isn't easy to do w/ IoError